### DISCUSSION TITLE
Bulk add tasks for a specific project/context

### DISCUSSION THREAD

**Discussion by befedi** [ransome1/sleek#314] [Ideas]
Hi,
I'm trying to use sleek to maintain my grocery list. This means I want to write down multiple items at a time for the same project or context. Creating a task for each item individually and having to add the project/context every time is quite tedious.  
 
On Android I am using "Todo.txt for Android" (https://play.google.com/store/apps/details?id=net.c306.ttsuper) which stays in the task adding mode until i close it (back button). Furthermore, if you filter for a specific project or context, this selection will be used to automatically tag all new tasks with the selection.

I wonder, if this would be an interesting feature for other users.

Regards,
befedi

**Comment by ransome1** [comment]
@befedi as a default behaviour I wouldn't like this. But we could introduce a keyboard/mouse combination to make use of your idea. For instance holding Ctrl and clicking Save could create a new todo and leave the modal open. And instead of clearing all previous values we can use the "Use as template" behaviour that you will find by right clicking on a todo in the list. How is this from a user experience perspective?

**Reply by befedi** [reply]
Hi @ransome1 I would also make this an optional behaviour. My suggestion in addition to your idea would be to press **Ctrl + <tbd>** or **Ctrl + Save** to save the todo and leave the modal open to create a new todo. 

The "Use as template" function is quite what I'm thinking of. The question is how to leave this function enabled while adding multiple todos. Maybe: having a toggle switch or similar in the creation modal with the title "Use as template" which you can activate while in the creation modal (or, if you use the "Use as template" function from the context menu, it would be already activated).

What do you think of this idea?

**Comment by aisbergde** [comment]
I like how this is realized on Android in Simpletask:
When adding new tasks, a "simple" text editor is opens, with the current context prefilled (+blablabla @foobar) and I just add new text before or after. And because it is a text editor, I can easy remove content. It is also possible to add new context or project using action buttons. And if I add a new line, the context is prefilled again.
IMHO this text editor in condition with action buttons is the very best way to insert new task, one ore more.

The current sleek way is not that comfortable for me, when adding one ore multiple task in the current context. Using the mouse to open drop down box to select the priority, this is just nice to have.

I would prefer a multi-line text editor, using auto completition, and only additional as helper the buttons and priority list, when someone forget the syntax.
